bot/cogs/eval.py

```diff
@@ -197,6 +222,16 @@ async def eval(self, ctx: Context, *, code: str) -> None:
         await self._eval(ctx, code)
 
 
+def find_nth_occurrence(string: str, substring: str, n: int) -> Optional[int]:
+    """Return index of `n`th occurrence of `substring` in `string`, or None if not found."""
+    index = 0
+    for _ in range(n):
+        index = string.find(substring, index+1)
+        if index == -1:
+            return None
+    return index
```

Comment:
This is generic enough that I think it might be better to keep it in utils. Might be useful somewhere else.

Reply:
Where would you suggest putting this, next to `pad_base64` in the utils' init or an another file?

Reply:
you really gonna ask me if I think it should be in the init after I just went on a whole rant about how nothing should be in the init? :grin: 

I don't even think `pad_base64` or `CogABCMeta` should be there. I never would've put it there myself. I wouldn't mind if you took this opportunity to clean that up as a bit of kaizen for this PR.

Reply:
These are really general helper functions, I'm not a big fan of say, `utils/utils` or `utils/helpers` as utils is fairly indescriptive by itself, but can't think of something better where to group them.

Reply:
`utils.py`, `helpers.py`, or even `general.py` are all fine by me. And of course, then you'd import them into the `__init__.py` (that's one thing you absolutely _should_ do in the init) so when someone wants to import them, they'd still just do `from bot.utils import whatever`.